### PR TITLE
View all Case Notes on Client; compose form dialog hooks

### PR DESCRIPTION
**Pull request by gigxz**
## Description

1. Add the ability to view Case Notes on the client dashboard. The tab is currently disabled because we need an API change in order to tell if the Case Notes feature is enabled for any projects.
2. Compose the "view" and "edit" form dialog hooks into one hook that handles both plus the associated state

## Type of change
- [x] New feature (adds functionality)
- [x] Code clean-up / housekeeping

## Checklist before requesting review
- [ ] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [ ] My code follows the style guidelines of this project (eslint)
- [ ] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/form/hooks/useViewEditRecordDialogs.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  evictCache?: VoidFunction;
+  localConstants?: LocalConstants;
+  maxWidth?: Breakpoint | false;
+}
+
+/**
+ * Helper for dialogs for Viewing and Editing/Deleting records.
+ * Only works for records that are submitted using SubmitForm.
+ * Often used alongside a table component.
+ */
```
Comment (gigxz) [inline src/modules/form/hooks/useViewEditRecordDialogs.tsx:27]: Just roughly pulling out a bunch of repeated logic into this hook.

File: src/modules/hmis/hmisUtil.ts [hunk 2]
```diff
@@ -369,6 +369,24 @@ export const assessmentDescription = (
 };
 
 export const eventReferralResult = (e: EventFieldsFragment) => {
+  if (e.probSolDivRrResult) {
+    if (e.probSolDivRrResult == NoYesMissing.Yes) {
```
Comment (ttoomey) [inline src/modules/hmis/hmisUtil.ts]: suggest always using strict equality '===' (maybe should be a linter rule)

**Review by ttoomey — APPROVED** [review]
Looks good!

File: src/modules/hmis/hmisUtil.ts (hunk 2 above)
Comment (gigxz) [reply src/modules/hmis/hmisUtil.ts]: whoops good call, added eqeqeq linter

**Pull request merged**